src/utils/search/searchWithDomSeek.ts

```diff
@@ -0,0 +1,157 @@
+        // console.log("start remainderToMoveForward: ", remainderToMoveForward);
+        // console.log("start iter.referenceNode.nodeValue: ", iter.referenceNode.nodeValue);
+        // console.log("start iter.referenceNode.nodeValue.length: ", iter.referenceNode.nodeValue.length);
+        // // if (remainderToMoveForward) {
+        // //     const previousRef = iter.referenceNode;
+        // //     actualMovedForward = domSeek(iter, remainderToMoveForward);
+        // //     console.log("start actualMovedForward (2): ", actualMovedForward);
+        // //     if (previousRef !== iter.referenceNode) {
+        // //         console.log("start previousRef !== iter.referenceNode: ", iter.referenceNode.nodeValue);
+        // //     }
+        // // }
+        // range.setStart(iter.referenceNode, remainderToMoveForward);
+
+        // moveForwardBy = matches[0].length;
+        // consumed += moveForwardBy;
+        // console.log("end moveForwardBy: ", moveForwardBy);
+        // actualMovedForward = domSeek(iter, moveForwardBy);
+        // remainderToMoveForward = moveForwardBy - actualMovedForward;
+        // console.log("end actualMovedForward: ", actualMovedForward);
+        // console.log("end remainderToMoveForward: ", remainderToMoveForward);
+        // if (range.startContainer !== iter.referenceNode) {
+        //     console.log("end RANGE SPANS ACROSS NODES:");
+        //     console.log("end iter.referenceNode.nodeValue: ", iter.referenceNode.nodeValue);
+        //     console.log("end iter.referenceNode.nodeValue.length: ", iter.referenceNode.nodeValue.length);
+        // }
+        // // if (remainderToMoveForward) {
+        // //     const previousRef = iter.referenceNode;
+        // //     actualMovedForward = domSeek(iter, remainderToMoveForward);
+        // //     console.log("end actualMovedForward (2): ", actualMovedForward);
+        // //     if (previousRef !== iter.referenceNode) {
+        // //         console.log("end previousRef !== iter.referenceNode: ", iter.referenceNode.nodeValue);
+        // //     }
+        // // }
+        // range.setEnd(iter.referenceNode, range.startContainer === iter.referenceNode ?
+        //     range.startOffset + remainderToMoveForward : remainderToMoveForward);
+
+        // // should be equal
+        // console.log("consumed (1): ", consumed);
+        // consumed = regexp.lastIndex;
+        // console.log("consumed (2): ", consumed);
+
+        if (!(doc as any).getCssSelector) {
+            (doc as any).getCssSelector = getCssSelector_(doc);
+        }
+        const rangeInfo = convertRange(
+            range,
+            (doc as any).getCssSelector,
+            (_node: Node) => ""); // computeElementCFI
+
+        searchResults.push({
+            textMatch: collapseWhitespaces(matches[0]),
+            textBefore,
+            textAfter,
+            rangeInfo,
+            href,
+            uuid: nanoid(),
```

Comment:
In my stress-tests I routinely generate many thousands (10-20) of search results, using large publications (many documents in the reading order), large documents (hundreds of "pages" each), and short frequently-occurring text tokens like "the" in the English language.

So, with that in mind, I am wondering whether the `uuid` of `ISearchResult` can simply be the concatenation of the stable short hash of `href`, and the incremental index `[0, n]` of the search result itself.

Rationale: I really like `nanoid` compared with `uuid.v4()`, but in this particular case I think we are wasting compute cycles for no good reason with randomly-generated IDs, because we can solidly rely on the ordered index of search results.

Thoughts?

Reply:
A parallel: in the React 'key' for iterated JSX fragments, we typically use the index, not a random number.

Reply:
fix with https://github.com/edrlab/thorium-reader/pull/1057/commits/0f68469fe7a8bfe5d6b29ad61194040d961968f0